hs_geographic_feature_resource/forms.py

```diff
@@ -0,0 +1,254 @@
+    fieldPrecision = forms.DecimalField(label='Field Precision', widget=forms.TextInput())
+
+    def __init__(self, allow_edit=True, res_short_id=None, element_id=None, *args, **kwargs):
+        super(FieldInformationForm, self).__init__(*args, **kwargs)
+        self.helper = FieldInformationForm(allow_edit, res_short_id, element_id, element_name='FieldInformation')
+        self.delete_modal_form = None
+        self.number = 0
+        self.delete_modal_form = None
+        self.allow_edit = allow_edit
+        self.errors.clear()
+
+        if not allow_edit:
+            for field in self.fields.values():
+                field.widget.attrs['readonly'] = True
+                field.widget.attrs['style'] = "background-color:white;"
+        # enabling all element fields for editing
+        # else:
+        #     self.fields['projection'].widget.attrs['readonly'] = True
+        #     self.fields['projection'].widget.attrs['style'] = "background-color:white;"
+        #     self.fields['units'].widget.attrs['readonly'] = True
+        #     self.fields['units'].widget.attrs['style'] = "background-color:white;"
+
+    def clean(self):
+        # modify the form's cleaned_data dictionary
+        super(FieldInformationForm, self).clean()
+        temp_cleaned_data = copy.deepcopy(self.cleaned_data)
+        is_form_errors = False
+        for limit in ('fieldName', 'fieldType', 'fieldTypeCode', 'fieldWidth', 'fieldPrecision'):
+            limit_data = temp_cleaned_data.get(limit, None)
+            if not limit_data:
+                self._errors[limit] = ["Data for %s is missing" % limit]
+                is_form_errors = True
+                del self.cleaned_data[limit]
+
+        if is_form_errors:
+            return self.cleaned_data
+
+        temp_cleaned_data['fieldName'] = str(temp_cleaned_data['fieldName'])
+        temp_cleaned_data['fieldType'] = str(temp_cleaned_data['fieldType'])
+        temp_cleaned_data['fieldTypeCode'] = str(temp_cleaned_data['fieldTypeCode'])
+        temp_cleaned_data['fieldWidth'] = str(temp_cleaned_data['fieldWidth'])
+        temp_cleaned_data['fieldPrecision'] = str(temp_cleaned_data['fieldPrecision'])
+
+        if 'fieldTypeCode' in temp_cleaned_data:
+            if len(temp_cleaned_data['fieldTypeCode']) == 0:
+                del temp_cleaned_data['fieldTypeCode']
+
+        if 'fieldPrecision' in temp_cleaned_data:
+            if len(temp_cleaned_data['fieldPrecision']) == 0:
+                del temp_cleaned_data['fieldPrecision']
+
+        if 'fieldWidth' in temp_cleaned_data:
+            if len(temp_cleaned_data['fieldWidth']) == 0:
+                del temp_cleaned_data['fieldWidth']
+
+        self.cleaned_data['value'] = copy.deepcopy(temp_cleaned_data)
+
+        if 'fieldName' in self.cleaned_data:
+            del self.cleaned_data['fieldName']
+        if 'fieldType' in self.cleaned_data:
+            del self.cleaned_data['fieldType']
```

Comment:
Is it absolutely necessary to remove existing fields from `cleaned_data`? It is not recommended.